### PR TITLE
Add the alerts for the google workspace data

### PR DESCRIPTION
**Pull request by hardikhdholariya**
Added the alerts for the google workspace data

### CODE REVIEW

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 1]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
 
-# ================
-# Google Workspace
-# ================
-[Google Workspace - Multiple Password Changes in Short Time Period]
+# =======================
+# Google Workspace (GWS)
+# =======================
+
+# TO DO - check the delay in data ingestion if it's realtime then consider _time otherwise go for _indextime
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4578]: Please check and update queries accordingly and then merge the PR.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gsuite_bulk_user_creation_deletion_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("CREATE_USER", "DELETE_USER")
+action.cyences_notable_event_action.attacker_search = | stats sum(total_actions) as count by admin
+action.cyences_notable_event_action.attacker_drilldown = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("CREATE_USER", "DELETE_USER") | rename "actor.email" as admin | search admin=$row.admin|s$ | stats count by user
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+# TO DO - Check if we can combine the above and below alert related to user.
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4617]: not for now

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+request.ui_dispatch_app = cyences_app_for_splunk
+request.ui_dispatch_view = search
+search = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("*USER*") NOT "event.name" IN ("CHANGE_CHROME_OS_USER_SETTING") _index_earliest=-61m@m _index_latest=-1m@m \
+| eval cyences_severity = case( in('event.name', `cs_gws_important_user_change_event`), "medium", true(), "low")  \
+| `cs_human_readable_time_format(_time, event_time)` \
+| `cs_gws_user_change_filter`  \
+| table event_time, cyences_severity, actor.email, event.name, user_email
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_user_change_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("*USER*")
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4646]: NOT "event.name" IN ("CHANGE_CHROME_OS_USER_SETTING")

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 4]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+| rex field=_raw "\"name\": \"old_value\", \"value\": \"(?<old_value>[^\"]*)\"" \
+| rex field=_raw "\"name\": \"new_value\", \"value\": \"(?<new_value>[^\"]*)\"" \
+| eval ModifiedProperties = info_setting." ::: ".old_value." --> ".new_value \
+| eval cyences_severity = case( in('event.name', `cs_gws_important_group_and_membership_change_event`), "medium", true(), "low")  \
+| `cs_human_readable_time_format(_time, event_time)` \
+| `cs_gws_group_change_filter`  \
+| table event_time, cyences_severity, actor.email, event.name, group_id, member_id, member_type, ModifiedProperties
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_group_change_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:groups_enterprise" NOT "event.name" IN ("*member*", "*user*")
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4683]: NOT "event.name" IN ("*member*", "*user*", "*join*")

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 5]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+[GWS - Enterprise Groups - GroupMembership Change/Update]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4688]: GWS - Enterprise Group Membership Change/Update

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 6]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+| `cs_gws_user_change_filter`  \
+| table event_time, cyences_severity, actor.email, event.name, user_email
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_user_change_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("*USER*")
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+[GWS - Enterprise Groups - Group Change/Update]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4651]: GWS - Enterprise Group Change/Update

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 7]
```diff
@@ -4571,10 +4571,189 @@ action.cyences_notable_event_action.products = Cisco IOS
...
+| `cs_gws_group_membership_change_filter`  \
+| table event_time, cyences_severity, actor.email, event.name, group_id, member_id, member_type, member_role
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_group_membership_change_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite` sourcetype="gws:reports:groups_enterprise" "event.name" IN ("*member*")
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+[GWS - Roles - Role Change/Update]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4724]: GWS - Role Change/Update

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 8]
```diff
@@ -4617,28 +4796,76 @@ counttype = number of events
...
-action.cyences_notable_event_action.attacker_drilldown = `cs_gsuite` sourcetype="gws:reports:admin" "event.name" IN ("CREATE_USER", "DELETE_USER") | rename "actor.email" as admin | search admin=$row.admin|s$ | stats count by user
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_successful_login_from_unusual_country_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite_login_events` "event.name"="login_success" `cs_public_ips(ipAddress)` | iplocation ipAddress
+action.cyences_notable_event_action.system_compromised_search = | stats count by user
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_gsuite_login_events` "event.name"="login_success" `cs_public_ips(ipAddress)` user=$row.user$ | iplocation ipAddress
 action.cyences_send_email_action = 1
 action.cyences_notable_event_action.products = Google Workspace
+
+
+[GWS - Suspicious Login Activity by User]
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4831]: Lets keep the **Google Workspace** as prefix for all alerts.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 9]
```diff
@@ -4617,28 +4796,76 @@ counttype = number of events
...
+action.cyences_notable_event_action = 1
+action.cyences_notable_event_action.param.filter_macro_name = cs_gws_suspicious_login_by_user_filter
+action.cyences_notable_event_action.contributing_events = `cs_gsuite_login_events` "event.name"="suspicious_login"
+action.cyences_notable_event_action.system_compromised_search = | stats count by user
+action.cyences_notable_event_action.system_compromised_drilldown = `cs_gsuite_login_events` "event.name"="suspicious_login" user=$row.user$
+action.cyences_send_email_action = 1
+action.cyences_notable_event_action.products = Google Workspace
+
+
+# TO DO - Add the DLP alert (regarding sharing the sensitive info) if it generates from the gws:reports:alert sourcetype
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4868]: Use TODO, so its easy to search, that's generic keyword for TODO item.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 10]
```diff
@@ -4617,28 +4796,76 @@ counttype = number of events
...
+# TO DO - Add the generic google alert from gws:reports:alert sourcetype
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4869]: good idea. If the data has severity in it then even amazing.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 11]
```diff
@@ -4617,28 +4796,76 @@ counttype = number of events
...
+# TO DO - Add the alerts related to the email logs
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4870]: for sure.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 12]
```diff
@@ -4617,28 +4796,76 @@ counttype = number of events
...
+# TO DO - Add the daily alert for the list of login failure happens in a day. (currently, data is not present)
```
Comment (VatsalJagani) [inline cyences_app_for_splunk/default/savedsearches.conf:4871]: No login data? We definitely need to collect login data.

**Review by VatsalJagani — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 12 above)
Comment (hardikhdholariya) [reply cyences_app_for_splunk/default/savedsearches.conf:4871]: Yes, we have login data for some event.name but we don't have login events with event.name=login_failure
I tried to login to the GWS admin account by entering invalid password but event is not generated. I asked the same to Rahaf.

**Review by VatsalJagani — APPROVED** [review]

**Pull request merged**